mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb

```diff
@@ -2384,12 +2384,29 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		cluster.SecurityPostureConfig = expandSecurityPostureConfig(v)
 	}
 
+  needUpdateAfterCreate := false
+
 	// For now PSC based cluster don't support `enable_private_endpoint` on `create`, but only on `update` API call.
 	// If cluster is PSC based and enable_private_endpoint is set to true we will ignore it on `create` call and update cluster right after creation.
 	enablePrivateEndpointPSCCluster := isEnablePrivateEndpointPSCCluster(cluster)
 	if enablePrivateEndpointPSCCluster {
 		cluster.PrivateClusterConfig.EnablePrivateEndpoint = false
-	}
+    needUpdateAfterCreate = true
+	}
+
+  enablePDCSI := isEnablePDCSI(cluster);
+  if !enablePDCSI {
+    // GcePersistentDiskCsiDriver cannot be disabled at cluster create, only on cluster update. Ignore on create then update after creation.
+    // If pdcsi is disabled, the config should be defined. But we will be paranoid and double-check.
+    needUpdateAfterCreate = true
+    if cluster.AddonsConfig == nil {
+      cluster.AddonsConfig = &container.AddonsConfig{}
+    }
+    if cluster.AddonsConfig.GcePersistentDiskCsiDriverConfig == nil {
+      cluster.AddonsConfig.GcePersistentDiskCsiDriverConfig = &container.GcePersistentDiskCsiDriverConfig{}
+    }
```

Comment:
Do we need these two checks here? I'd assume we wouldn't reach these two conditions since we'd have `enablePDCSI = true` if `cluster.AddonsConfig == nil || cluster.AddonsConfig.GcePersistentDiskCsiDriverConfig == nil`? Or I could be missing some cases.

Reply:
There are several other addons besides pdcsi. So it definitely could be the case that addons config != nil but the pdcsi config is nil.

However, as mentioned in the comment above I'm being paranoid because if pdcsi is not enabled, the config must have been explicitly defined.

But it seems an easy check to make and who knows how the code might change in the future.

Reply:
Yeah, that makes sense. I was just curious since these checks are under `if !enablePDCSI`. But let's just keep them just in case. Thanks!